src/plugins/telemetryTable/collections/TableRowCollection.js

```diff
@@ -85,28 +85,49 @@ define(['lodash', 'EventEmitter'], function (_, EventEmitter) {
     }
 
     sortAndMergeRows(rows) {
-      const sortedRowsToAdd = this.sortCollection(rows);
+      const sortedRows = this.sortCollection(rows);
 
       if (this.rows.length === 0) {
-        this.rows = sortedRowsToAdd;
+        this.rows = sortedRows;
 
         return;
       }
 
-      const firstIncomingRow = sortedRowsToAdd[0];
-      const lastIncomingRow = sortedRowsToAdd[sortedRowsToAdd.length - 1];
+      const firstIncomingRow = sortedRows[0];
+      const lastIncomingRow = sortedRows[sortedRows.length - 1];
       const firstExistingRow = this.rows[0];
       const lastExistingRow = this.rows[this.rows.length - 1];
 
       if (this.firstRowInSortOrder(lastIncomingRow, firstExistingRow) === lastIncomingRow) {
-        this.rows = [...sortedRowsToAdd, ...this.rows];
+        this.insertOrUpdateRows(sortedRows, true);
       } else if (this.firstRowInSortOrder(lastExistingRow, firstIncomingRow) === lastExistingRow) {
-        this.rows = [...this.rows, ...sortedRowsToAdd];
+        this.insertOrUpdateRows(sortedRows, false);
       } else {
-        this.mergeSortedRows(sortedRowsToAdd);
+        this.mergeSortedRows(sortedRows);
       }
     }
 
+    getInPlaceUpdateIndex(row) {
+      const inPlaceUpdateKey = row.inPlaceUpdateKey;
+      if (!inPlaceUpdateKey) {
+        return false;
```

Comment:
This should return -1, no? because `(false > -1) === true` (See line 144)

Reply:
oh ya! Thanks.